pkg-manager/resolve-dependencies/src/updateProjectManifest.ts

```diff
@@ -23,25 +17,17 @@ export async function updateProjectManifest (
   if (!importer.manifest) {
     throw new Error('Cannot save because no package.json found')
   }
-  const specsToUpsert = opts.directDependencies
+  const specsToUpsert: PackageSpecObject[] = opts.directDependencies
     .filter((rdd, index) => importer.wantedDependencies[index]?.updateSpec)
     .map((rdd, index) => {
       const wantedDep = importer.wantedDependencies[index]!
-      return resolvedDirectDepToSpecObject({
-        ...rdd,
-        isNew:
-        wantedDep.isNew,
-        currentPref: wantedDep.pref,
-        preserveNonSemverVersionSpec: wantedDep.preserveNonSemverVersionSpec,
-        // For git-protocol dependencies that are already installed locally, there is no normalizedPref unless do force resolve,
-        // so we use pref in wantedDependency here.
-        normalizedPref: rdd.normalizedPref ?? (isGitHostedPkgUrl((rdd.resolution as TarballResolution).tarball ?? '') ? wantedDep.pref : undefined),
-      }, importer, {
+      return {
+        alias: rdd.alias,
         nodeExecPath: wantedDep.nodeExecPath,
-        pinnedVersion: wantedDep.pinnedVersion ?? importer.pinnedVersion ?? 'major',
-        preserveWorkspaceProtocol: opts.preserveWorkspaceProtocol,
-        saveWorkspaceProtocol: opts.saveWorkspaceProtocol,
-      })
+        peer: importer['peer'],
```

Comment:
Why not just `importer.peer`?

```suggestion
        peer: importer.peer,
```